src/backup.rs

```diff
@@ -139,3 +202,43 @@ impl Drop for RestoreOptions {
         }
     }
 }
+
+#[test]
+fn backup_restore() {
+    use db::DBVector;
+    // create backup
+    let path = "_rust_rocksdb_backup_restore_test";
```

Comment:
@rohitjoshi you can use `util::DBPath` struct here

Reply:
I followed existing test cases patterns from `db.rs`.  `util::DBPath` is under `tests` directory.

Reply:
It will be cool if you'd invoke `DB::Destroy` to remove tmp directories `_rust_rocksdb_backup_restore_test`, `_rust_rocksdb_backup_path` and `_rust_rocksdb_restore_from_backup_path` after test complete.

Reply:
@rohitjoshi I think this is the last change we need. Invoking `DB::Destroy` will make sure the test database will be cleaned up at the end of the test run.  As you noted, `util::DBPath` is only available in the `tests` crate.  I'm working on a PR that, among other things, moves `DBPath` into the main crate and calls it `TemporaryDBPath`.  Once we merge your PR into `master`, I can incorporate it into my PR and updated it to use `TemporaryDBPath`. But for now, `DB::Destroy` will suffice to keep what's on `master` from leaving temporary databases around.

Thanks for all your work on this!